CMakeLists.txt

```diff
@@ -505,9 +505,7 @@ target_compile_definitions(re-objs PRIVATE ${RE_DEFINITIONS})
 
 target_include_directories(re-objs PRIVATE .)
 target_include_directories(re-objs PRIVATE include)
-if(USE_OPENSSL)
-  target_include_directories(re-objs PUBLIC ${OPENSSL_INCLUDE_DIR})
-endif()
+target_link_libraries(re-objs PRIVATE ${CRYPTO_LIBS})
```

Comment:
`re-objs` are only OBJECTS and should not be linked.

Reply:
Then how to make it compile using proper include directories both for OpenSSL and MbedTLS?
Without this change, compilation fails on MacOS (MbedTLS installed via brew):

```
$ cmake -H. -Bbuild
-- The C compiler identification is AppleClang 12.0.0.12000032
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found MbedTLS: /usr/local/lib/libmbedtls.dylib  
in mbedtls /usr/local/lib/libmbedtls.dylib
-- Looking for backtrace
-- Looking for backtrace - found
...
$ cmake --build build
/Users/dmitryilyin/git/re/src/md5/wrap.c:11:10: fatal error: 'mbedtls/md5.h' file not found
#include <mbedtls/md5.h>
         ^~~~~~~~~~~~~~~
1 error generated.
gmake[2]: *** [CMakeFiles/re-objs.dir/build.make:902: CMakeFiles/re-objs.dir/src/md5/wrap.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:104: CMakeFiles/re-objs.dir/all] Error 2
```

Reply:
Like the removed include for OPENSSL_INCLUDE_DIR:

`target_include_directories(re-objs PRIVATE ${MBEDTLS_INCLUDE_DIRS})`

Reply:
Changed as suggested.

But as for me, it looks wordy and old CMake-fashioned